### ISSUE TITLE
Some issues with the repo

### ISSUE THREAD

**Issue by Dandelight** [niranjankrishna-acad/Training-Vision-Transformers-with-Only-2040-Images#1]
Although I your loss function design may be good, there are several (possibly critical) problems about your implementation.

1. You have even no accuracy evaluated in your code.
2. The permutations of classes the dataloader produces are not random -- the code just iterate over the dataset linearly, refering to a `counter` global variable to determine which class an image belongs to and then update the `counter` variable. This leads to the problem of having a batch of images of the same class, which is problematic because it tends to make the training process unstable.
3. `(counter)/17` does not work with the dataset https://www.robots.ox.ac.uk/~vgg/data/flowers/17/ , since the images are grouped every 80 images, so it should be written as `(counter)/80`.
4. Global variables like `counter` are highly prone to cause problems, especially in a parallel context. Try using a class member or a function closure.
